core/src/components/cat-tabs/cat-tabs.tsx

```diff
@@ -0,0 +1,107 @@
+import { Component, h, Element, State, Watch, Listen, Host, Prop } from '@stencil/core';
+
+/**
+ * @part tab - The header of the tab.
+ */
+@Component({
+  tag: 'cat-tabs',
+  styleUrl: 'cat-tabs.scss',
+  shadow: true
+})
+export class CatTabs {
+  private tabs: HTMLCatTabElement[] = [];
+  private buttons: HTMLCatButtonElement[] = [];
+
+  @Element() hostElement!: HTMLElement;
+
+  @State() activeTabId?: string;
+
+  /**
+   * The ID of the active tab.
+   */
+  @Prop() activeTab = '';
+
+  /**
+   * The tabs alignment
+   */
+  @Prop() tabsAlign: 'left' | 'center' | 'justify' = 'left';
+
+  @Watch('activeTabId')
+  onActiveTabChanged(newActiveTab: string): void {
+    const activeTab = this.tabs.find(value => value.id === newActiveTab);
+    activeTab?.click();
+  }
+
+  componentWillLoad(): void {
+    this.tabs = Array.from(this.hostElement.querySelectorAll('cat-tab'));
+    if (this.tabs.length > 0) {
+      let nextUniqueId = 0;
+      this.tabs.forEach(tab => {
+        if (!tab.id) tab.id = `cat-checkbox-${nextUniqueId++}`;
+      });
+      this.activeTabId = this.activeTab;
+    }
+  }
+
+  componentDidRender(): void {
+    this.buttons.forEach((button: HTMLCatButtonElement) => {
+      const element = button.shadowRoot?.querySelector('button');
+      element?.setAttribute('tabindex', button.buttonId === this.activeTabId ? '0' : '-1');
+    });
+  }
+
+  @Listen('keydown')
+  onKeydown(event: KeyboardEvent): void {
+    if (['ArrowDown', 'ArrowUp', 'ArrowRight', 'ArrowLeft'].includes(event.key)) {
```

Comment:
nice implementation. Works. Did you check if one could use the focus trap library (that we already using), can be used here?

Reply:
in this case it would not be possible to use the focus trap. It only works with the Tab and Shift+Tab keys, not the arrow keys.